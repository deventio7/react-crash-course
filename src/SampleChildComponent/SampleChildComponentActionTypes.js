const SAMPLE_CHILD_COMPONENT_SUBMIT = 'SAMPLE_CHILD_COMPONENT_SUBMIT';

export default {
  SAMPLE_CHILD_COMPONENT_SUBMIT
};