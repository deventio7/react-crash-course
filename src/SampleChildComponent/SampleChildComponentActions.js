import SampleChildComponentActionTypes from './SampleChildComponentActionTypes';

const updateStoredText = (text) => {
  return {
    type: SampleChildComponentActionTypes.SAMPLE_CHILD_COMPONENT_SUBMIT,
    payload: text
  }
}

export default {
	updateStoredText
}