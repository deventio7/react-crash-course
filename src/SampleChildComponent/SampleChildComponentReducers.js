import SampleChildComponentActionTypes from './SampleChildComponentActionTypes';

const initialState = {
	storedText: ''
}

const sampleChildSubmit = (state = initialState, action) => {
	switch (action.type) {
		case SampleChildComponentActionTypes.SAMPLE_CHILD_COMPONENT_SUBMIT:
			return {
				...state,
				storedText: action.payload
			};
		default:
			return state;
	}
}

export default {
  sampleChildSubmit
};