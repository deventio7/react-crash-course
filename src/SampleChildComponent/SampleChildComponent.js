import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SampleChildComponentActions from './SampleChildComponentActions';

export class SampleChildComponent extends Component {
  constructor() {
    super();
    this.state = {
      inputText: ''
    }
  };

  submit = () => {
    this.props.actions.submit(this.state.inputText);
  }

  updateInputValue = (event) => {
    this.setState({
      inputText: event.target.value
    });
  }

  render = () => {
    console.log('Rendering SampleChildComponent');
    return (
      <div>
        SampleChildComponent<br/>
        <input type="text" onChange={ this.updateInputValue }/>
        <button onClick={ this.submit }>submit</button>
      </div>
    );
  }
}

/* istanbul ignore next */
const mapDispatchToProps = (dispatch) => {
  return {
    actions: bindActionCreators(SampleChildComponentActions, dispatch)
  }
}

SampleChildComponent.propTypes = {
  actions: PropTypes.object.isRequired
}

export default connect(undefined, mapDispatchToProps)(SampleChildComponent);
