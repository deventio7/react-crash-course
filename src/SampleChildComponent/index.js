import SampleChildComponent from './SampleChildComponent';

export default SampleChildComponent;