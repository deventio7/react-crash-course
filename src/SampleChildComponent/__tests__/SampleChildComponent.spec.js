import React from 'react';
import { shallow } from 'enzyme';
import { SampleChildComponent } from '../SampleChildComponent';

describe('SampleChildComponent', () => {
  const basicProps = {
    actions: {
      submit: jest.fn()
    }
  }

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders and matches snapshot', () => {
    const wrapper = shallow(<SampleChildComponent {...basicProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('updateInputValue changes state', () => {
    const wrapper = shallow(<SampleChildComponent {...basicProps} />).instance();
    jest.spyOn(wrapper, 'setState');
    wrapper.updateInputValue({
      target: {
        value: 'TEST VALUE'
      }
    });
    expect(wrapper.setState).toBeCalledWith({ inputText: 'TEST VALUE' });
    expect(wrapper.state.inputText).toBe('TEST VALUE');
  });

  it('submit calls action', () => {
    const wrapper = shallow(<SampleChildComponent {...basicProps} />).instance();
    wrapper.state.inputText = 'TEST VALUE';
    wrapper.submit();
    expect(basicProps.actions.submit).toBeCalledWith('TEST VALUE');
  });
});
