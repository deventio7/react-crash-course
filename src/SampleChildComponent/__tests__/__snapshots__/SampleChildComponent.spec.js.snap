// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleChildComponent renders and matches snapshot 1`] = `
<div>
  SampleChildComponent
  <br />
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    onClick={[Function]}
  >
    submit
  </button>
</div>
`;
